File: solutions/2149. Rearrange Array Elements by Sign/2149.py
class Solution:
  def rearrangeArray(self, nums: list[int]) -> list[int]:
    ans = []
    pos = []
    neg = []

    for num in nums:
      (pos if num > 0 else neg).append(num)

    for p, n in zip(pos, neg):
      ans += [p, n]

    return ans

File: solutions/2149. Rearrange Array Elements by Sign/2149.cpp
class Solution {
 public:
  vector<int> rearrangeArray(vector<int>& nums) {
    vector<int> ans;
    vector<int> pos;
    vector<int> neg;

    for (const int num : nums)
      (num > 0 ? pos : neg).push_back(num);

    for (int i = 0; i < pos.size(); ++i) {
      ans.push_back(pos[i]);
      ans.push_back(neg[i]);
    }

    return ans;
  }
};
